eng/scripts/Verify-PreviewDependency.ps1

```diff
@@ -4,14 +4,20 @@ param (
   [string]$PackagePath
 )
 
+."$PSScriptRoot\..\common\scripts\SemVer.ps1"
+
 $packagePaths = Get-ChildItem -Path "$PackagePath/*/package.json"
 
 foreach ($packagePath in $packagePaths) {
   $pkgContent = Get-Content $packagePath | ConvertFrom-Json
   foreach ($dependency in $pkgContent.dependencies.PSObject.properties) {
     $depVer = $dependency.Value
-    if (($depVer -match "\D0.\d.\d") -or ($depVer -match "\D\d.\d.\d-\w")) {
-      throw "Found dependency on preview/pre-release package $($dependency.Name) with version range $depVer. Please pin dependencies to preview packages."
+    if ($depVer -match "\D\d.\d.\d") {
+      $baseVer = $depVer.replace("~","").replace("^","")
+      $version = [AzureEngSemanticVersion]::new($baseVer)
+      if ($version.IsPrerelease) {
+        throw "Found dependency on preview/pre-release package $($dependency.Name) with version range $depVer. Please pin dependencies to preview packages."
```

Comment:
```suggestion
        throw "Found dependency on preview/pre-release package '$($dependency.Name)' with version range '$depVer'. Please pin dependencies to preview packages."
```

I always try to wrap values in quotes, because sometimes bugs can cause them to be empty and it's not obvious from the log they are missing when there are no quotes.